Source/SwiftLintFramework/Extensions/Array+SwiftLint.swift

```diff
@@ -24,3 +24,27 @@ extension Array where Element: NSTextCheckingResult {
         return map { $0.range }
     }
 }
+
+extension Array where Element: Equatable {
+    var unique: [Element] {
+        var uniqueValues: [Element] = []
+        forEach { item in
+            if !uniqueValues.contains(item) {
+                uniqueValues += [item]
+            }
+        }
+        return uniqueValues
+    }
+}
+
+public extension SequenceType {
+    // swiftlint:disable line_length
+    func categorize<U: Hashable>(@noescape by transform: Generator.Element -> U) -> [U: [Generator.Element]] {
```

Comment:
By the way, SourceKitten already has implementations for these, in case you want to compare:

https://github.com/jpsim/SourceKitten/blob/0.15.2/Source/SourceKittenFramework/ClangTranslationUnit.swift#L16-L33